src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,36 @@
 package core.basesyntax.service;
 
+import core.basesyntax.InvalidUserException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int AGE_THRESHOLD = 18;
+    private static final int MIN_NUMBER_OF_CHARS = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws InvalidUserException {
+        if (user == null
+                || user.getAge() == null
+                || user.getLogin() == null
+                || user.getPassword() == null) {
+            throw new InvalidUserException("User's data is invalid. Try again");
```

Comment:
I think it's better to split all these conditions for separate 'if statement' and for each condition add more informative exception message. For "user == null -  User can't be null", etc..